src/connect.ts

```diff
@@ -76,6 +78,8 @@ async function connectWithFailFast(uri: string, client: MongoClient, logger: Con
     if (failEarlyClosePromise !== null) {
       await failEarlyClosePromise;
       throw failedConnections.values().next().value; // Just use the first failure.
+    } else if (isAtlas(uri)) {
+      (err as Error).message = 'It looks like this is a MongoDB Atlas cluster. Please ensure that your IP whitelist allows connections from your network.';
```

Comment:
Why are we replacing the entire error message here, and why are we not checking what type of error we saw?

Reply:
Yeah, seems that according to the doc: `For Atlas “IP Whitelist” = “Network Access List”`